app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewHolder.kt

```diff
@@ -217,12 +217,15 @@ class AttendeeViewHolder(private val binding: ItemAttendeeBinding) : RecyclerVie
     }
 
     fun checkValidFields(): Boolean {
+        var valid = true
         requiredList.forEach {
-            if (!it.checkEmpty() ||
-                (it.inputType == InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS && !it.checkValidEmail()) ||
-                (it.inputType == InputType.TYPE_TEXT_VARIATION_URI && !it.checkValidURI())) return false
+            if (!it.first.checkEmpty(it.second) ||
+                (it.first.inputType == InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS &&
+                    !it.first.checkValidEmail(it.second)) ||
+                (it.first.inputType == InputType.TYPE_TEXT_VARIATION_URI &&
+                    !it.first.checkValidURI(it.second))) valid = false
```

Comment:
this if condition looks too complicated, why don't you simplify it with helper methods

Reply:
updated